mono/mini/debugger-agent.c

```diff
@@ -5489,6 +5562,27 @@ process_single_step_inner (DebuggerTlsData *tls, gboolean from_signal, uint64_t
 	process_event (EVENT_KIND_STEP, jinfo_get_method (ji), il_offset, ctx, events, suspend_policy);
 #else
 	Il2CppSequencePointC* sequence_pt = tls->il2cpp_context.sequencePoints[tls->il2cpp_context.frameCount - 1];
+
+	/*
+	* This could be in ss_update method, but mono_find_next_seq_point_for_native_offset is pretty expensive method,
+	* hence we prefer this check here.
+	*/
+	if (ss_req->user_assemblies)
+	{
+		gboolean found = FALSE;
+		for (int k = 0; ss_req->user_assemblies[k]; k++)
+		{
+			if (ss_req->user_assemblies[k] ==  VM_IMAGE_GET_ASSEMBLY(VM_CLASS_GET_IMAGE(VM_METHOD_GET_DECLARING_TYPE(*sequence_pt->method))))
+			{
+				found = TRUE;
+				break;
+			}
+		}
+
+		if (!found)
+			return;
```

Comment:
We probably don't need `found` in this method, as it will always return, independent of the value of `found`.

Reply:
If found is false it returns before calling process_event() at the bottom of the function.  This code was copied from Mono's code for doing assembly filtering while single stepping.

Reply:
I completely missed that. Sorry, ignore me.